torch/csrc/jit/operator.cpp

```diff
@@ -146,7 +158,11 @@ struct SchemaParser {
         L.next(); // [
         L.next(); // ]
         value = ListType::create(value);
-        alias_info = AliasInfo(parseAliasAnnotation(writes), {std::move(alias_info)});
+        auto container = parseAliasAnnotation();
+        if (container && alias_info) {
```

Comment:
Does all this optional handling logic mean that alias_info might not mirror the structure of the contained type? That will create hellish logic trying to understand when alias information existed (Imagine what will happen when you have alias analysis on tuple and do not unconditionally add them as contained types). The previous version was careful to make sure that AliasInfo mirrored the struct exactly.

Reply:
That makes sense. I can fix it up so that if the type is composite we unconditionally report alias information (even if it's empty).

Reply:
actually this is easier to fix up on top the alias analysis PR, so I'm going to do it there.